state/unit.go

```diff
@@ -1454,12 +1458,32 @@ func (u *Unit) SetStatus(unitStatus status.StatusInfo) error {
 }
 
 // OpenedPortRanges returns a UnitPortRanges object that can be used to query
+// and/or mutate the port ranges opened by the unit.
+func (u *Unit) OpenedPortRanges() (UnitPortRanges, error) {
+	if u.ShouldBeAssigned() {
+		return u.openedPortRangesForMachine()
+	}
+	isSidecar, err := u.IsSidecar()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if u.isCaas() && !isSidecar {
+		return nil, errors.NotSupportedf("open/close ports for %q", u.ApplicationName())
+	}
+	return u.openedPortRangesForApplication()
```

Comment:
This is confusing because it's not "for application". It's just "what ports has this unit opened".